### PR TITLE
Replace deprecated command with environment file

### LINKED ISSUE #1635: Replace deprecated `set-output` command with environment file
**Describe the issue**
In workflow, `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information, see: [https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/)

### PR DESCRIPTION
**Pull request by jongwooo**
## Description

Resolve  #1635 

Update workflows to use environment file instead of deprecated `set-output` command. 
For more information, see: [https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/)

I found the workflow files that use `set-output` command through the following command:

```bash
$ find . -name '*.yml' -o -name '*.yaml' | xargs egrep '\bset-output\b'
```

**AS-IS**

```yml
run: |
  echo "::set-output name=go-build::$(go env GOCACHE)"
  echo "::set-output name=go-mod::$(go env GOMODCACHE)"
```

**TO-BE**

```yml
run: |
  echo "go-build=$(go env GOCACHE)" >> $GITHUB_OUTPUT
  echo "go-mod=$(go env GOMODCACHE)" >> $GITHUB_OUTPUT
shell: bash
```

### CODE REVIEW

**Review by torredil — APPROVED** [review]
Thanks @jongwooo /lgtm

**Review by ConnorJC3 — CHANGES_REQUESTED** [review]
/hold

This PR breaks the Windows unit tests

**Review by ConnorJC3 — APPROVED** [review]
/remove-hold
/lgtm

Thanks! I'll leave it open a few days to make sure other maintainers have a chance to review, but this revision is good as far as I can see (you can ignore the Windows e2e test failure, it's due to a known ongoing issue).

**Pull request merged**
